WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -93,6 +93,14 @@ def index
       end
     end
 
+    begin
+      from_date = params[:from_date] && Date.parse(params[:from_date])
+      @competitions = @competitions.where("start_date >= ?", from_date) if from_date
+      to_date = params[:to_date] && Date.parse(params[:to_date])
+      @competitions = @competitions.where("end_date =< ?", to_date) if to_date
+    rescue ArgumentError
```

Comment:
Ah, it differs subtly. It requires the format to be `'%Y-%m-%d'`, whereas `#parse` handles many of them.